### PR TITLE
Assorted FreeBSD fixes

### PR DESCRIPTION
**Pull request by arrowd**
* Target version: master 

### Summary

With these patches I was able to get everything compiled and linked. The build still fails, so no runtime testing yet.

```
sudo  cap_fowner,cap_mknod,cap_sys_chroot=ep loolforkit && sudo  cap_sys_admin=ep loolmount
sudo: loolforkit: command not found
```

### Checklist

- [x] Code is properly formatted
- [x] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [x] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Comment by vmiklos** [comment]
One of the commits miss the signed-off-by line in the commit message, that's why the DCO check fails. Could you please do an interactive rebase and amend the problematic commit(s)? Thanks.

I didn't notice anything obviously problematic in the actual changes.

File: net/Socket.cpp [hunk 1]
```diff
@@ -698,6 +702,16 @@ int Socket::getPid() const
...
     return creds.pid;
+#else
+    struct xucred creds;
+    socklen_t credSize = sizeof(struct xucred);
+    if(getsockopt(_fd, 0, LOCAL_PEERCRED, &creds, &credSize) < 0)
+    {
+        LOG_TRC("Failed to get pid via peer creds on " << _fd << ' ' << strerror(errno));
+        return -1;
+    }
+    return creds.cr_pid;
```
Comment (myfreeweb) [inline net/Socket.cpp:713]: Condition this to `__FreeBSD__ >= 13`, there was no `cr_pid` before https://reviews.freebsd.org/rS348419 which wasn't MFC'd

File: net/Socket.cpp [hunk 2]
```diff
@@ -690,6 +693,7 @@ std::shared_ptr<Socket> ServerSocket::accept()
 
 int Socket::getPid() const
 {
+#ifndef __FreeBSD__
```
Comment (myfreeweb) [inline net/Socket.cpp:696]: `SOL_SOCKET SO_PEERCRED` isn't standard either, so this should be specific to Linux. (And OpenBSD IIRC?)

File: net/Ssl.cpp [hunk 3]
```diff
@@ -167,7 +167,11 @@ void SslContext::lock(int mode, int n, const char* /*file*/, int /*line*/)
 
 unsigned long SslContext::id()
 {
+#ifdef __linux
     return syscall(SYS_gettid);
+#else
```
Comment (myfreeweb) [inline net/Ssl.cpp:172]: Maybe `#elif defined(__FreeBSD__) pthread_getthreadid_np()` if thread ID vs process ID matters

File: common/Util.cpp [hunk 4]
```diff
@@ -525,10 +527,12 @@ namespace Util
         int res = setpriority(PRIO_PROCESS, pid, prio);
         LOG_TRC("Lowered kit [" << (int)pid << "] priority: " << prio << " with result: " << res);
 
+#ifdef __linux
         // rely on Linux thread-id priority setting to drop this thread' priority
         pid_t tid = getThreadId();
         res = setpriority(PRIO_PROCESS, tid, prio);
         LOG_TRC("Lowered own thread [" << (int)tid << "] priority: " << prio << " with result: " << res);
+#endif
```
Comment (myfreeweb) [inline common/Util.cpp:535]: `#elif defined(__FreeBSD__)` could something like `rtprio_thread(RTP_SET, pthread_getthreadid_np(), {..idle or normal rtprio..})` work?

File: common/FileUtil.cpp [hunk 5]
```diff
@@ -18,6 +18,9 @@
 #include <sys/vfs.h>
 #elif defined IOS
 #import <Foundation/Foundation.h>
+#elif defined __FreeBSD__
```
Comment (myfreeweb) [inline common/FileUtil.cpp:21]: huh, defined can work without `(` parens `)`? TIL

**Review by mmeeks — COMMENTED** [review]
This is fine for now; but I wonder if we should have an abstract class with system bits in it like this - we could then use that for stubbing / testing separately which would be useful =) saves some #ifdefs too perhaps.

File: net/Socket.cpp (hunk 1 above)
Comment (arrowd) [reply net/Socket.cpp:713]: There is nothing we can do in `<= 12` case, only fail with `#error`, that is, different error message.

**Comment by arrowd** [comment]
Can this be merged in?

**Comment by mrkara** [comment]
@arrowd I still see 2 unresolved conversations. Anything to do about them?

Anyway, since it looks generally OK, and already got positive comments from @mmeeks and @vmiklos , I'll merge this PR if no objections/concerns raised until 16:30 UTC+3. :)

**Comment by tml1024** [comment]
I am not sure if I am even able to build the iOS app from the master branch at the moment, will check. (Presumably still against cp-6.4 of core?) If I can build the iOS app from the master branch (and even run the result), I will check whether this PR breaks it.

**Comment by tml1024** [comment]
Hmm, but this PR is for a branch from another repo even, sigh. No idea how to handle that.

**Review by tml1024 — APPROVED** [review]
Does not seem to break the iOS app any more than how it already is broken when built from the master branch, so OK from my POV.

**Comment by mmeeks** [comment]
Sorry for the delay ! =) Any joy getting it working nicely ?

**Pull request merged**